pkg/handler/handler.go

```diff
@@ -96,26 +99,95 @@ type ReadmeUtgaveFilter struct {
 	First  *int32
 }
 
-func (r *resolver) ReadmeUtgaver(filter *ReadmeUtgaveFilter) ([]ReadmeUtgave, error) {
-	if filter == nil {
-		return r.readmes, nil
+func getLink(name string) string {
+	return fmt.Sprintf("https://storage.googleapis.com/%s/%s", bucketName, name)
+}
+
+func getRegexMatches(name string) []string {
+	re := regexp.MustCompile(`(?P<Year>\d{4})-(?P<Utgave>\d{2})`)
+
+	if !re.MatchString(name) {
+		return nil
 	}
-	var filteredReadmes []ReadmeUtgave
-	for _, r := range r.readmes {
-		if filter.Year != nil && *filter.Year != r.Year {
-			continue
+	return re.FindStringSubmatch(name)[1:]
+}
+
+// getReadmes fetches readmes from the storage bucket via a storage.Client.
+// query is optional for filtering objects via object prefix (path).
+// ex. "2018" fiters for all objects with a path beginning with "2018"
+func getReadmes(ctx context.Context, client storage.Client, query string) ([]ReadmeUtgave, error) {
+	pdfQuery := &storage.Query{Prefix: fmt.Sprintf("pdf/%s", query)}
+	imageQuery := &storage.Query{Prefix: fmt.Sprintf("images/%s", query)}
+
+	bkt := client.Bucket(bucketName)
+	pdfIt := bkt.Objects(ctx, pdfQuery)
+	imageIt := bkt.Objects(ctx, imageQuery)
+	var err error
+	var readmes []ReadmeUtgave
+	for {
+		pdfAttrs, errr := pdfIt.Next()
+		if errr == iterator.Done {
+			break
+		}
+		imageAttrs, errr := imageIt.Next()
+
+		if errr == iterator.Done {
+			break
 		}
-		if filter.Utgave != nil && *filter.Utgave != r.Utgave {
+
+		// Object may be a directory
+		matches := getRegexMatches(pdfAttrs.Name)
+		if matches == nil {
 			continue
 		}
-		filteredReadmes = append(filteredReadmes, r)
-		if filter.First != nil && len(filteredReadmes) == int(*filter.First) {
+
+		year, errr := strconv.ParseInt(matches[0], 10, 32)
+		utgave, errr := strconv.ParseInt(matches[1], 10, 32)
```

Comment:
this one too

Reply:
Edit: not this one :laughing:

Reply:
Hmm, yeah. I see the error handling here is **not** :100: